doc/nrf/ug_matter_overview_architecture.rst

```diff
@@ -0,0 +1,142 @@
+It could also define input from a specific PIN keyboard UI, reaction on a specific LED on the door bolt model, and so on.
+
+.. _ug_matter_architecture_overview_dm:
+
+Data Model
+==========
+
+The Data Model layer describes how to separate data into abstract logical blocks that help maintain the application in a comfortable manner.
+The layer operates regardless of the business logic, and so for example it is identical for each door lock type of device.
+
+.. figure:: images/matter_components_DM.svg
+   :alt: Data Model layer overview
+
+   Data Model layer overview
+
+Nodes
+   Each device is composed of one or more nodes, which are complete implementations of a Matter application functionality on a single stack.
+   Nodes are identifiable with a unique network address on a single network.
+   They can communicate directly with other nodes in the network.
+
+Endpoints
+   Each node is composed of one or more endpoints, which contain feature sets of single device functionalities.
+   For example, in a voice-controlled door lock device, one endpoint could include the feature set for operating the bolt, while the other endpoint could include the feature set for handling a temperature sensor.
+
+   .. note::
+      Endpoint 0 is always reserved for Matter's Utility Clusters.
+      This is the only mandatory endpoint for each Matter device.
+
+Clusters
+   Endpoints are composed of one or more clusters, which group attributes, commands and events that together make up each single feature in a set.
+   For example, in an endpoint for operating a door bolt, one cluster could group attributes for moving the bolt to open or closed position, while other could group attributes for controlling alarms in case the door is opened in unauthorized way.
+
+   Clusters can be of the following types:
+
+   * Server -- responsible for holding values for Attributes, Commands, and Events.
+   * Client -- responsible for performing interactions with other Server Clusters.
+
+   The supported Matter application clusters are described in the Application Clusters specification, available to the `Connectivity Standards Alliance`_ members.
+   Sets of clusters on one or more endpoint can form a device type, that is an officially defined collection of requirements that is conformant with the Device Library specification.
+
+Attributes
+   Attributes are data entities that represent a physical quantity or state.
+   They are stored in the Matter device's memory, but can also be calculated dynamically on demand.
+
+Commands
+   Commands are actions that can be used to trigger some behavior on other devices.
+   For example, in a door lock device the lock door command can be used to trigger such action on a physical device.
+
+Events
+   Events are a type of attributes that communicate device state changes.
```

Comment:
The unique feature of events is that you can read multiple historical records for the same event. E.g. if you have a HardwareFault event, and it happens that several hardware faults occur, then if you read the event you will receive a list of events (the count depends on the size of the event buffer). On the other hand, if you read an attribute, you will always receive the current value of that attribute. Therefore, events != reportable attributes and subscribing to an event is not the same as subscribing to an attribute change. Having said that, I'm not sure if saying that an event is a type of attribute is correct, but I'm not sure who came up with that definition and based on which sources :). To me they are similar to attributes but they don't store historical occurrences of a certain... event instead of the current value.

Reply:
@Damian-Nordic, hanks for the clarification. So maybe we can simply skip the mention about "type of attributes" and have sth like:

```suggestion
   Events are a mechanism that communicates device state changes.
```